flink/v1.17/flink/src/main/java/org/apache/iceberg/flink/source/IcebergSource.java

```diff
@@ -76,6 +78,7 @@
   private final ScanContext scanContext;
   private final ReaderFunction<T> readerFunction;
   private final SplitAssignerFactory assignerFactory;
+  private final Comparator<IcebergSourceSplit> splitComparator;
```

Comment:
`Comparator` interface is not extended from `Serializable`. Although JDK doc recommends implementing comparator to implement serializable, it is not not part of the interface contract.

We may want to introduce a `SerializableComparator` interface like `SerializableSupplier`.